### PR TITLE
vpn

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
See https://github.com/golemfactory/yagna/issues/1393 for a dependency list

### CODE REVIEW

File: core/vpn/src/device.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    queue: &'a mut VecDeque<Vec<u8>>,
+}
+
+impl<'a> phy::TxToken for TxToken<'a> {
+    fn consume<R, F>(self, _timestamp: time::Instant, len: usize, f: F) -> smoltcp::Result<R>
+    where
+        F: FnOnce(&mut [u8]) -> smoltcp::Result<R>,
+    {
+        let mut buffer = Vec::with_capacity(len);
+        buffer.resize(len, 0);
```
Comment (maaktweluit) [inline core/vpn/src/device.rs:74]: this line seems to do nothing: `buffer` was just initialised with the same`len` as capacity.

I propose to:
- Remove the line
- Init the `buffer` => `with_capacity(0)`

**Review by maaktweluit — COMMENTED** [review]
Nice work!

Will continue the review tomorrow morning :)

File: core/vpn/src/device.rs (hunk 1 above)
Comment (mfranciszkiewicz) [reply core/vpn/src/device.rs:74]: It fills the buffer of capacity `len` with zeroes. Before that, `buffer.len()` is equal to 0.

Playground
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=7492e3da91e3adc5a7fcfc4e63c89c23

Comment (Wiezzel) [reply core/vpn/src/device.rs:74]: How about this?
```rust
let mut buffer = vec![0; len];
```

Comment (maaktweluit) [reply core/vpn/src/device.rs:74]: Ahhhhh, i was thinking about this option later in the evening yesterday.. TIL!

Thank you! Disregard my comment :)

I like @Wiezzel 's proposal here too, but its not as "broken" ( read sub-obtimal ) as i thought ;)

**Review by maaktweluit — APPROVED** [review]
LGTM!

**Review by nieznanysprawiciel — DISMISSED** [review]

**Pull request merged**